### PR TITLE
refactor: create two generic methods and use them to avoid repetition

### PR DESCRIPTION
**Pull request by ftonato**
## What kind of change does this PR introduce?

- refactor: create two generic methods and use them to avoid repetition

**Note:** There is still a lot of repetition, especially on routes, but this pull-request can help remove duplication

## What is the new behavior?

Both behaviours remain the same.

## Additional context

> The open-source community is incredible!

**Note:** Sorry for the extensive pull-request...

### CODE REVIEW

**Comment by ftonato** [comment]
Sorry for the failed tests, I didn't have a DB locally and didn't want to install just to run the tests ^^

> Suggestion, maybe you can make the tests run with a docker :)

**Comment by inian** [comment]
Yup, there is a local docker for db already. Will push up the docs once we finish our launch for today :D

**Comment by inian** [comment]
Updated the [Readme](https://github.com/supabase/storage-api/blob/master/README.md). Let me know if something can be made clearer or easier to get started!

**Comment by ftonato** [comment]
It’s clearer, now I just need to create an account to have AWS credentials

File: src/routes/bucket/createBucket.ts [hunk 1]
```diff
@@ -45,26 +45,18 @@ export default async function routes(fastify: FastifyInstance) {
         owner = await getOwner(jwt)
       } catch (err) {
         console.log(err)
-        return response.status(400).send({
-          statusCode: '400',
-          error: err.message,
-          message: err.message,
-        })
+        return response.status(400).send()
```
Comment (inian) [inline src/routes/bucket/createBucket.ts:48]: @ftonato can you add back the error bodies here?

File: src/routes/bucket/createBucket.ts [hunk 2]
```diff
@@ -45,26 +45,18 @@ export default async function routes(fastify: FastifyInstance) {
...
 
       const { name: bucketName } = request.body
-      let id = request.body.id
-      if (!id) {
-        //by default set the id as the name of the bucket
-        id = bucketName
-      }
+
+      // IMPORTANT: by default set the id as the name of the bucket
+      const id = request.body.id || bucketName
```
Comment (inian) [inline src/routes/bucket/createBucket.ts:54]:
```suggestion
      const id = request.body.id ?? bucketName
```

**Comment by inian** [comment]
Thanks @ftonato..few small changes in the review, I will merge first though.

File: src/routes/bucket/createBucket.ts (hunk 1 above)
Comment (ftonato) [reply src/routes/bucket/createBucket.ts:48]: Sorry for the delay, I saw that you already corrected it, thanks =)

**Pull request merged**